ConsistencyManager/DataStructures/WeakArray.swift

```diff
@@ -117,14 +117,19 @@ public struct WeakArray<T: AnyObject>: ArrayLiteralConvertible {
 
 // MARK: MutableCollectionType Implementation
 
-extension WeakArray: MutableCollectionType {
+extension WeakArray: MutableCollection {
 
     // Required by SequenceType
-    public func generate() -> IndexingGenerator<WeakArray<T>> {
+    public func makeIterator() -> IndexingIterator<WeakArray<T>> {
         // Rather than implement our own generator, let's take advantage of the generator provided by IndexingGenerator
-        return IndexingGenerator<WeakArray<T>>(self)
+        return IndexingIterator<WeakArray<T>>(_elements: self)
     }
 
+    public func index(after i: Int) -> Int {
+        //TODO: modify the code to return the correct value
```

Comment:
Oh...yeah, I see now. Yeah, I think this is the right implementation too. I think it's ok to remove this TODO. If tests pass, then I think it's safe to assume this is correct.